subsys/net/lib/websocket/websocket.c

```diff
@@ -818,178 +863,103 @@ int websocket_recv_msg(int ws_sock, uint8_t *buf, size_t buf_len,
 	}
 #endif /* CONFIG_NET_TEST */
 
-	/* If we have not received the websocket header yet, read it first */
-	if (!ctx->header_received) {
-#if defined(CONFIG_NET_TEST)
-		size_t input_len = MIN(ctx->tmp_buf_len - ctx->tmp_buf_pos,
-				       test_data->input_len);
+	ctx->data_buf_count = 0;
+	ctx->data_buf_size = buf_len;
+	ctx->data_buf = buf;
```

Comment:
I wonder if those really need to be a part of the `struct websocket_context` given they are re-initialized on each `websocket_recv_msg()` call. They could be local variables, the parser function could accept additional params.

Reply:
It is good idea, i add `struct websocket_payload_buffer`.

Reply:
Now `struct websocket_buffer` is used.